src/trio/_tests/test_testing_raisesgroup.py

```diff
@@ -140,23 +256,59 @@ def test_catch_unwrapped_exceptions() -> None:
         raise ExceptionGroup("", [ValueError()])
 
     # with allow_unwrapped=False (default) it will not be caught
-    with pytest.raises(ValueError, match="^value error text$"):
+    with pytest.raises(
+        AssertionError,
+        match=wrap_escape(
+            "Raised exception group did not match: ValueError('value error text') is not an exception group, but would match with `allow_unwrapped=True`",
+        ),
+    ):
         with RaisesGroup(ValueError):
             raise ValueError("value error text")
 
-    # allow_unwrapped on it's own won't match against nested groups
-    with pytest.raises(ExceptionGroup):
+    # allow_unwrapped on its own won't match against nested groups
+    # TODO: even with the flatten_subgroups suggestion I feel like this message is a bit confusing. It is not obvious at all that
+    # the mentioned ExceptionGroup (bar) is a sub-exception (of "foo"). Same can be said for some other cases.
+    with pytest.raises(
+        AssertionError,
+        match=wrap_escape(
+            "Raised exception group did not match: ExceptionGroup('bar', [ValueError()]) is not of type 'ValueError'\n"
+            "Did you mean to use `flatten_subgroups=True`?",
```

Comment:
Could say "inner ExceptionGroup(..." and Did you mean to use `flatten_subgroups=True`[ instead of `allow_unwrapped=True`]?" I guess?